### PR TITLE
DON-596 – transfer funds tip handling

### PR DESCRIPTION
**Pull request by NoelLH**
* ask Identity to return any existing tip balances awaiting bank transfer funding
* if present in GBP, acknowledge any existing tip
* and don't prompt for another, including the Gift Aid form group

### CODE REVIEW

File: src/app/identity.service.ts [hunk 1]
```diff
@@ -81,19 +81,24 @@ export class IdentityService {
       person);
   }
 
-  get(id: string, jwt: string): Observable<Person> {
+  get(id: string, jwt: string, withTipBalances: boolean): Observable<Person> {
```
Comment (bdsl) [inline src/app/identity.service.ts:84]: It's sort of OK in the IDE, but when we read code that calls this in github we'd have to remember what the third param means - if we just see false or true that doesn't tell us much. So could we make it
```suggestion
  get(id: string, jwt: string, {withTipBalances}: {withTipBalances: boolean}): Observable<Person> {
```

The repetition of `withTipBalances` is a bit ugly, but then the calling code gets a lot clearer:
```diff
-    this.identityService.get(id, jwt, false).subscribe(
+    this.identityService.get(id, jwt, {withTipBalances: false}).subscribe(
```

File: src/app/person.model.ts [hunk 2]
```diff
@@ -14,6 +14,12 @@ export interface Person {
      */
     cash_balance?: { [currencyCode: string]: number };
 
+    /**
+     * The total of donor fund form-created pending tips for the given Person, which will be fulfilled
```
Comment (bdsl) [inline src/app/person.model.ts:18]:
```suggestion
     * The total of donor fund form-created pending tips for the given Person, counted in minor-currency units i.e. pence which will be fulfilled
```

**Review by bdsl — COMMENTED** [review]
Added inline comment. Also wondering if we should turn off the new section of the form about permission to send them emails when we're not asking for a tip, since it isn't relevant.

File: src/app/identity.service.ts (hunk 1 above)
Comment (bdsl) [reply src/app/identity.service.ts:84]: Or we could make false the default and then this PR doesn't need to change any of the callers except the one that passes true:
```suggestion
  get(id: string, jwt: string, {withTipBalances = false}: {withTipBalances?: boolean} = {}): Observable<Person> {
```

File: src/app/identity.service.ts [hunk 3]
```diff
@@ -102,7 +107,7 @@ export class IdentityService {
       return of(null);
     }
 
-    return this.get(idAndJWT.id, idAndJWT.jwt);
+    return this.get(idAndJWT.id, idAndJWT.jwt, false);
```
Comment (NoelLH) [inline src/app/identity.service.ts:110]:
```suggestion
    return this.get(idAndJWT.id, idAndJWT.jwt);
```

File: src/app/donation-start/donation-start-container/donation-start-container.component.ts [hunk 4]
```diff
@@ -87,7 +87,7 @@ export class DonationStartContainerComponent implements AfterViewInit, OnInit{
       return;
     }
 
-    this.identityService.get(id, jwt).subscribe(
+    this.identityService.get(id, jwt, false).subscribe(
```
Comment (NoelLH) [inline src/app/donation-start/donation-start-container/donation-start-container.component.ts:90]:
```suggestion
    this.identityService.get(id, jwt).subscribe(
```

File: src/app/transfer-funds/transfer-funds.component.ts [hunk 5]
```diff
@@ -342,17 +342,24 @@ export class TransferFundsComponent implements AfterContentInit, OnInit {
 
   private loadAuthedPersonInfo(id: string, jwt: string) {
     this.isLoading = true;
-    this.identityService.get(id, jwt).subscribe((person: Person) => {
+    this.identityService.get(id, jwt, true).subscribe((person: Person) => {
```
Comment (NoelLH) [inline src/app/transfer-funds/transfer-funds.component.ts:345]:
```suggestion
    this.identityService.get(id, jwt, {withTipBalances: true}).subscribe((person: Person) => {
```

File: src/app/donation-start/donation-start-form/donation-start-form.component.ts [hunk 6]
```diff
@@ -2093,7 +2093,7 @@ export class DonationStartFormComponent implements AfterContentChecked, AfterCon
 
             if (this.donor?.id) {
               const jwt = this.identityService.getJWT() as string;
-              this.identityService.get(this.donor?.id, jwt).subscribe((person: Person) => {
+              this.identityService.get(this.donor?.id, jwt, false).subscribe((person: Person) => {
```
Comment (NoelLH) [inline src/app/donation-start/donation-start-form/donation-start-form.component.ts:2096]:
```suggestion
              this.identityService.get(this.donor?.id, jwt).subscribe((person: Person) => {
```

**Comment by NoelLH** [comment]
> Added inline comment. Also wondering if we should turn off the new section of the form about permission to send them emails when we're not asking for a tip, since it isn't relevant.

Yep good point – that's hopefully done in https://github.com/thebiggive/donate-frontend/pull/1362/commits/ad5e8a325cd9601ae0c0b5cae7236e5ee6641061

**Review by bdsl — APPROVED** [review]

**Pull request merged**